From: Mark Wielaard <mark@klomp.org>
Date: Thu, 21 Feb 2019 23:28:01 +0000 (+0100)
Subject: libelf: Remove unused internal __elf[32|64]_msize functions.
X-Git-Tag: elfutils-0.177~43
X-Git-Url: https://sourceware.org/git/?p=elfutils.git;a=commitdiff_plain;h=be8080bdd746ac2b07fb0bcad23a9677844bb200

libelf: Remove unused internal __elf[32|64]_msize functions.

Those functions were intended for ELF versions where the memory and
file sizes of data structures are different. They were never used
because libelf depends on the file and memory sizes being equal
(otherwise using mmap wouldn't work).

Signed-off-by: Mark Wielaard <mark@klomp.org>
---

diff --git a/libelf/elf32_fsize.c b/libelf/elf32_fsize.c
index 0f468de5..139f4a91 100644
--- a/libelf/elf32_fsize.c
+++ b/libelf/elf32_fsize.c
@@ -58,5 +58,3 @@ elfw2(LIBELFBITS, fsize) (Elf_Type type, size_t count, unsigned int version)
 	  * __libelf_type_sizes[0][ELFW(ELFCLASS,LIBELFBITS) - 1][type]);
 #endif
 }
-#define local_strong_alias(n1, n2) strong_alias (n1, n2)
-local_strong_alias (elfw2(LIBELFBITS, fsize), __elfw2(LIBELFBITS, msize))
diff --git a/libelf/libelfP.h b/libelf/libelfP.h
index bc9a404d..3a96a3b0 100644
--- a/libelf/libelfP.h
+++ b/libelf/libelfP.h
@@ -444,16 +444,6 @@ extern const uint_fast8_t __libelf_type_aligns[ELFCLASSNUM - 1][ELF_T_NUM]
    be ELF_T_BYTE.  */
 extern Elf_Type __libelf_data_type (Elf *elf, int sh_type) internal_function;
 
-/* The libelf API does not have such a function but it is still useful.
-   Get the memory size for the given type.
-
-   These functions cannot be marked internal since they are aliases
-   of the export elfXX_fsize functions.*/
-extern size_t __elf32_msize (Elf_Type __type, size_t __count,
-			     unsigned int __version);
-extern size_t __elf64_msize (Elf_Type __type, size_t __count,
-			     unsigned int __version);
-
 
 /* Create Elf descriptor from memory image.  */
 extern Elf *__libelf_read_mmaped_file (int fildes, void *map_address,
